src/PersistentFile.js

```diff
@@ -42,12 +33,11 @@ class PersistentFile extends EventEmitter {
     const json = {
       size: this.size,
       path: this.path,
-      name: this.name,
-      type: this.type,
+      newName: this.newName,
+      mime: this.mime,
       mtime: this.lastModifiedDate,
       length: this.length,
       filename: this.filename,
-      mime: this.mime,
     };
     if (this.hash && this.hash !== '') {
```

Comment:
`this.hashAlgorithm`

Reply:
Should I split into hashResult and hashAlgorithm ?

Reply:
`this.hash` is okay to remain, cuz it will be the actual hex, while `hashAlgorithm` is an option.
previously `hash` meant both the option and the final resulted hex and was a bit messy